src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {

```

Comment:
Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. You have correctly parameterized `MachineProducer` with `Excavator`, which is a subtype of `Machine`. Good job!